qemu/tests/convert_to_virtual_disk.py

```diff
@@ -0,0 +1,44 @@
+import logging
+
+from provider.qemu_img_utils import setup_loop_dev
+from provider.qemu_img_utils import free_loop_dev
+
+from avocado import fail_on
+from avocado.utils import process
+
+from virttest import data_dir
+from virttest.qemu_storage import QemuImg
+
+
+def run(test, params, env):
+    """
+    Convert a image to a virtual block device.
+    1. Create source and middle images.
+    2. Setup loop device with the middle image.
+    3. Convert the source image to the loop device.
+    :param test: Qemu test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+    src_img = params["images"].split()[0]
+    mid_img = params["images"].split()[-1]
+    root_dir = data_dir.get_data_dir()
+    source = QemuImg(params.object_params(src_img), root_dir, src_img)
+    middle = QemuImg(params.object_params(mid_img), root_dir, mid_img)
+    mid_filename = middle.image_filename
+
+    logging.info("Create the test image files.")
+    source.create(source.params)
+    middle.create(middle.params)
+
+    logging.info("Setup target loop device via 'losetup'.")
+    target = setup_loop_dev(mid_filename)
+    params["image_name_target"] = target
+
+    logging.debug("Convert from %s to %s with cache mode none.",
+                  source.image_filename, mid_filename)
+    fail_on((process.CmdError,))(source.convert)(
+        params.object_params(src_img), root_dir, cache_mode="none")
+
+    logging.info("Clean the loop device.")
+    free_loop_dev(target)
```

Comment:
`free_loop_dev(target)` maybe not be executed if test failed.

Reply:
Updated, thanks.